osquery/tables/system/posix/last.cpp

```diff
@@ -11,19 +11,46 @@
 
 #include <osquery/core/core.h>
 #include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
 
 namespace osquery {
 namespace tables {
 
 namespace impl {
 
+std::string typeNameForType(const utmpx& ut) {
+  switch (ut.ut_type) {
+  case EMPTY:
+    return "empty";
+  case RUN_LVL:
+    return "run-level";
+  case BOOT_TIME:
+    return "boot-time";
+  case NEW_TIME:
+    return "new-time";
+  case OLD_TIME:
+    return "old-time";
+  case INIT_PROCESS:
+    return "init-process";
+  case LOGIN_PROCESS:
+    return "login-process";
+  case USER_PROCESS:
+    return "user-process";
+  case DEAD_PROCESS:
+    return "dead-process";
+  }
+
+  return "";
+}
+
 void genLastAccessForRow(const utmpx& ut, QueryData& results) {
   if (ut.ut_type == USER_PROCESS || ut.ut_type == DEAD_PROCESS) {
     Row r;
     r["username"] = TEXT(ut.ut_user);
     r["tty"] = TEXT(ut.ut_line);
     r["pid"] = INTEGER(ut.ut_pid);
     r["type"] = INTEGER(ut.ut_type);
+    r["type_name"] = TEXT(typeNameForType(ut));
```

Comment:
Could name this `type`, and renamed the old to `type_num`. It'd be a breaking change, but arguably makes the table more useful.

Reply:
The current approach is workable, it makes sense and isn't a breaking change.